bin/scripts/continious-data-population.js

```diff
@@ -0,0 +1,1183 @@
+                "sum": [50, 10000],
+                "dur": [60, 600]
+            },
+            "Settings Changed": {
+                "segments": {"Setting": ["Address", "Payment method"]}
+            },
+        }
+    },
+    {
+        "_id": "defaultGaming",
+        "name": "Gaming",
+        "isDefault": true,
+        "up": {
+            "Experience Points": [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
+        },
+        "events": {
+            "Login": {
+                "segments": {"Method": ["Facebook", "Google", "Email"]}
+            },
+            "Level Up": {
+                "segments": {"Level": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], "Used Level Pass": [true, false]},
+                "dur": [60, 600]
+            },
+            "Purchase": {
+                "segments": {"Item": ["Level Pass", "Lucky Item", "Item Storage Upgrade"]},
+                "sum": [1, 100]
+            },
+            "Share Score": {
+                "segments": {"To": ["Facebook", "Twitter", "Instagram"]}
+            },
+            "Invite Friends": {}
+        }
+    }
+];
+var app = null, template = null, pluginList = null;
+var users = [], user = null;
+var stats = {u: 0, s: 0, x: 0, d: 0, e: 0, r: 0, b: 0, c: 0, p: 0};
+
+var plugins = require('../../plugins/pluginManager.js'),
+    request = require('request'),
+    Chance = require('chance');
+
+
+function writeMsg(msg) {
+    process.stdout.write(JSON.stringify(msg) + '\n');
+    process.exit(1);
+}
+/**
+ * Checks the correctness of the required parameters.
+ **/
+async function readDbParameters() {
+    try {
+        const db = await plugins.dbConnection("countly");
+        app = await db.collection('apps').findOne({'_id': db.ObjectID(APP_ID)});
+        pluginList = await db.collection('plugins').find({}).toArray();
+        if (POPULATOR_TEMPLATE_ID.length > 18) {
+            template = await db.collection('populator_templates').findOne({'_id': db.ObjectID(POPULATOR_TEMPLATE_ID)});
+        }
+        else { // default template
+            defaultTemplates.forEach(function(item) {
+                if (item._id === POPULATOR_TEMPLATE_ID) {
+                    template = item;
+                }
+            });
+            if (!template) {
+                db.close();
+                writeMsg('INVALID POPULATOR_TEMPLATE_ID ', POPULATOR_TEMPLATE_ID);
+            }
+        }
+        if (!app) {
+            db.close();
+            writeMsg('INVALID APP_ID ', APP_ID);
+        }
+        if (!template) {
+            db.close();
+            writeMsg('INVALID POPULATOR_TEMPLATE_ID ', POPULATOR_TEMPLATE_ID);
+        }
+        db.close();
+    }
+    catch (error) {
+        clearInterval(interval);
+        writeMsg('There was an error while checking db parameters', error);
+    }
+}
+
+readDbParameters();
+
+var interval = setInterval(async function() {
+    users = [];
```

Comment:
You call this function on each interval. Why are you resetting users on each call?